pkgs/development/libraries/cxxopts/default.nix

```diff
@@ -0,0 +1,27 @@
+{ cmake, fetchFromGitHub, icu, lib, pkg-config, stdenv, enableUnicodeHelp ? false }:
+
+stdenv.mkDerivation rec {
+  name = "cxxopts";
+  version = "2.2.1";
+
+  src = fetchFromGitHub {
+    owner = "jarro2783";
+    repo = name;
+    rev = "v${version}";
+    sha256 = "0d3y747lsh1wkalc39nxd088rbypxigm991lk3j91zpn56whrpha";
+  };
+
+  buildInputs = lib.optional enableUnicodeHelp [ icu.dev ];
```

Comment:
```suggestion
  buildInputs = lib.optional enableUnicodeHelp [ icu ];
```

Reply:
With this change I get:

```
-- Checking for module 'icu-uc'
--   No package 'icu-uc' found
CMake Error at /nix/store/zqnyzfkyyys15bfgc9sr1z0098fqq6zv-cmake-3.19.2/share/cmake-3.19/Modules/FindPkgConfig.cmake:553 (message):
  A required package was not found
Call Stack (most recent call first):
  /nix/store/zqnyzfkyyys15bfgc9sr1z0098fqq6zv-cmake-3.19.2/share/cmake-3.19/Modules/FindPkgConfig.cmake:741 (_pkg_check_modules_internal)
  CMakeLists.txt:62 (pkg_check_modules)
```

when building enableUnicodeHelp

Reply:
buildInputs should choose the correct variant. Not sure what wrong here.

Reply:
Is there any documentation on this choosing process?

Reply:
Couldn't find one.